wordpress-mobile/WordPress-iOS
# Better error-message logic for post uploads (for page reusability).

This PR is the first step towards leveraging our Offline Post messages for use in Offline Pages.

Changes:

- Relocated offline post messaging (so that the code is cleaner and it's easier to reuse our existing logic for pages).
- Implemented offline page messaging.  This is still not wired / no need to test this.
- Restructures some unit tests so we will be able to extend them for pages easily.
- Reorganizes some unit tests to add cases we weren't testing in offline posting.

### To test:

While offline, try a combination of saving a draft, publishing a post, private post, scheduling, submitting for review.

This should be mostly covered by the proposed unit tests, so make sure to dedicate some time to reviewing the unit tests.

### PR submission checklist:
